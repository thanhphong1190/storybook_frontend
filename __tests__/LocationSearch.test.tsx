import renderer from 'react-test-renderer'
import {
  LocationSearchForm,
  LocationSearchFormProps
} from '../src/components/search/LocationSearchForm'

const props: LocationSearchFormProps = {
  locationOptions: [
    { label: '2560', value: '2536' },
    { label: '2015', value: '2015' },
    { label: '2156', value: '2156' },
    { label: '2064', value: '2064' },
    { label: '2527', value: '2527' },
    { label: 'Ambarvale', value: 'Ambarvale' },
    { label: 'ANNANGOVE', value: 'ANNANGOVE' },
    { label: 'Alexander Heights', value: 'Alexander Heights' },
    { label: 'Alkimos', value: 'Alkimos' },
    { label: 'Ashfield', value: 'Ashfield' }
  ],
  searchLabel: 'Find your local Plus Fitness',
  searchPlaceHolder: 'Enter a suburb or postcode',
  searchButtonText: 'GO'
}

it('renders correctly', () => {
  const tree = renderer.create(<LocationSearchForm {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
