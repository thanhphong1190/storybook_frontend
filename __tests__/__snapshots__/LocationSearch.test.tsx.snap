// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="SearchFormstyled__LocationSearchWrapper-sc-1lsr3ed-0 fCmkeW"
>
  <label
    className="SearchFormstyled__SearchLabel-sc-1lsr3ed-2 kGYlKu"
    htmlFor="searchText"
  >
    Find your local Plus Fitness
  </label>
  <div
    className="SearchFormstyled__SearchFormGroup-sc-1lsr3ed-3 gsSrke"
  >
    <div
      className="SearchFormstyled__AsyncSelectStyled-sc-1lsr3ed-1 dGHfvT css-2b097c-container"
      id="searchText"
      onKeyDown={[Function]}
    >
      <div
        className=" css-yk16xz-control"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <div
          className=" css-g1d714-ValueContainer"
        >
          <div
            className=" css-1wa3eu0-placeholder"
          >
            Enter a suburb or postcode
          </div>
          <div
            className="css-b8ldur-Input"
          >
            <div
              className=""
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-autocomplete="list"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                disabled={false}
                id="react-select-2-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "boxSizing": "content-box",
                    "color": "inherit",
                    "fontSize": "inherit",
                    "label": "input",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "1px",
                  }
                }
                tabIndex="0"
                type="text"
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </div>
        </div>
        <div
          className=" css-1hb7zxy-IndicatorsContainer"
        >
          <span
            className=" css-1okebmr-indicatorSeparator"
          />
          <div
            aria-hidden="true"
            className=" css-tlfecz-indicatorContainer"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <svg
              aria-hidden="true"
              className="css-6q0nyr-Svg"
              focusable="false"
              height={20}
              viewBox="0 0 20 20"
              width={20}
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <button
      className="Button-lxhj0a-0 eQpnsG"
    >
      GO
    </button>
  </div>
</div>
`;
